rct229/rulesets/ashrae9012019/section11/section11rule7.py

```diff
@@ -0,0 +1,151 @@
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_swh_equipment_type import (
+    get_swh_equipment_type,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_swh_uses_associated_with_each_building_segment import (
+    get_swh_uses_associated_with_each_building_segment,
+)
+from rct229.schema.schema_enums import SchemaEnums
+from rct229.utils.assertions import getattr_
+from rct229.utils.jsonpath_utils import find_exactly_one_with_field_value
+from rct229.utils.pint_utils import ZERO
+
+SERVICE_WATER_HEATING_SPACE = SchemaEnums.schema_enums[
+    "ServiceWaterHeatingSpaceOptions2019ASHRAE901"
+]
+
+
+def get_swh_bats_and_swh_use():
+    """
+    This function will be removed once the function is developed
+    """
+
+    return {}
+
+
+class Section11Rule7(RuleDefinitionListIndexedBase):
+    """Rule 7 of ASHRAE 90.1-2019 Appendix G Section 11 (Service Water Heating)"""
+
+    def __init__(self):
+        super(Section11Rule7, self).__init__(
+            rmds_used=produce_ruleset_model_description(
+                USER=False, BASELINE_0=True, PROPOSED=True
+            ),
+            each_rule=Section11Rule7.RMDRule(),
+            index_rmd=BASELINE_0,
+            id="11-7",
+            description="Except in buildings that will have no service water heating loads, the service water heating system type in the baseline building design shall be as specified in Table G3.1.1-2 for each building area type in the proposed design.",
+            ruleset_section_title="Service Water Heating",
+            standard_section="Table G3.1 #11, baseline column, a & b",
+            is_primary_rule=True,
+            list_path="ruleset_model_descriptions[0]",
+        )
+
+    class RMDRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section11Rule7.RMDRule, self).__init__(
+                rmds_used=produce_ruleset_model_description(
+                    USER=False,
+                    BASELINE_0=True,
+                    PROPOSED=True,
+                ),
+            )
+
+        def is_applicable(self, context, data=None):
+            rmd_p = context.PROPOSED
+
+            swh_bats_and_uses_p = get_swh_bats_and_swh_use(rmd_p)
+
+            for swh_bat in swh_bats_and_uses_p:
+                for swh_use_id in swh_bats_and_uses_p[swh_bat]:
+                    swh_uses = find_exactly_one_with_field_value(
+                        "$.buildings[*].building_segments[*].zones[*].spaces[*].service_water_heating_uses[*]",
+                        "id",
+                        swh_use_id,
+                        rmd_p,
+                    )
+                    if (
+                        getattr_(swh_uses, "service_water_heating_uses", "use")
+                        > ZERO.VOLUME
+                        or swh_bat != SERVICE_WATER_HEATING_SPACE.PARKING_GARAGE
+                    ):
+                        return True
+                    else:
+                        return False
```

Comment:
My general feeling is the RDS did not explain where the applicability/evaluation happens. The note says `Applicability is determined at the building segment level, but the rule is evaluated for all SWH equipment serving the building segment.` However, the logic is on `swh_bat`.

In that case, I suggest we use evaluate at `swh_bat` level.
This would require restructure of the context to only include the returned value from `get_SWH_components_associated_with_each_SWH_bat`. See Section5rule14 for the implementation.

Reply:
Confirmed with Karen, this rule should be evaluated at building area type level - we should update the note in the RDS to reflect this change.

Reply:
Addressed.